fahrschein/src/main/java/org/zalando/fahrschein/ReaderManager.java

```diff
@@ -0,0 +1,10 @@
+package org.zalando.fahrschein;
+
+import org.zalando.fahrschein.domain.Subscription;
+
+import java.util.Optional;
+
+@FunctionalInterface
+public interface ReaderManager {
```

Comment:
Could you also add a default implementation of this interface, similar to the `DefaultErrorHandler` that is used instead of the anonymous functions in `NakadiClientBuilder` and the tests?